src/Propel/Runtime/ActiveQuery/Criteria.php

```diff
@@ -723,13 +724,23 @@ public function getNewCriterion(string $column, $value = null, $comparison = sel
             // something like $c->add('foo like ?', '%bar%', PDO::PARAM_STR);
             return new RawCriterion($this, $column, $value, $comparison);
         }
+        if ($value instanceof Criteria && $comparison === self::EQUAL) {
+            $comparison = InQueryCriterion::IN;
+        }
```

Comment:
Changed it so that `$comparison` defaults to `null`, now we can change it to `IN` only when no operator was given. That way, if `WHERE col = (SELECT...)` is ever implemented (currently causes an exception), it can still be built by passing in an explicit `EQUAL` operator along with the query.

I don't think it would make sense to generally use `=` as the default operator when a query is given, because from what I know, `WHERE col = (SELECT...)` is just a special case of `IN` where the subquery returns exactly one element. So while both give you the same result, `=` does not always work, while `IN` does.
And if no operator is given, I think we should pick the option that always works, not the special case.
What do you think?

Reply:
I have generalized the IN query, so that `filterBy()` accepts a query as value along with any operator. So now it is possible to build statements like `WHERE col = (SELECT...)`